client.go

```diff
@@ -356,6 +356,131 @@ func (c *Client) logf(format string, v ...interface{}) {
 	}
 }
 
+// Broadcast a message to all members in a Group
+func (c *Client) Broadcast(timeout time.Duration, g *Group, msg interface{}) (BroadcastResult, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	return c.broadcast(ctx, cancel, g, msg)
+}
+
+// BroadcastC (broadcast) a message to all members in a Group. The context can be used to control
+// cancellations or timeouts
+func (c *Client) BroadcastC(ctx context.Context, g *Group, msg interface{}) (BroadcastResult, error) {
+	cont, cancel := context.WithCancel(ctx)
+	defer cancel()
+	return c.broadcast(cont, cancel, g, msg)
+}
+
+func (c *Client) broadcast(ctx context.Context, cancel context.CancelFunc, g *Group, msg interface{}) (BroadcastResult, error) {
+	res := make(BroadcastResult)
+	receivers := g.Members()
+
+	var broadcastErr error
+	mu := new(sync.Mutex)
+	wg := new(sync.WaitGroup)
+	for _, rec := range receivers {
+		wg.Add(1)
+		go func(receiver string) {
+			defer wg.Done()
+			resp, err := c.RequestC(ctx, receiver, msg)
+			switch {
+			case err == nil && g.fastest:
+				// if this request was successful and the group is configured to Fastest,
+				// then cancel the context so other requests are terminated
+				cancel()
+			case err != nil && !g.fastest:
+				broadcastErr = ErrBroadcast
+			}
+
+			mu.Lock()
+			res[receiver] = &Result{
+				Err: err,
+				Val: resp,
+			}
+			mu.Unlock()
+		}(rec)
+	}
+	wg.Wait()
+	return res, broadcastErr
+}
+
+// Group defines a group of actors. This struct is primarily used for
+// broadcasting messages to all actors in a Group.
+type Group struct {
+	fastest bool
+	members []string
+}
+
+// NewListGroup creates a new Group
+func NewListGroup(members ...string) *Group {
+	return &Group{
+		members: members,
+	}
+}
+
+// Members returns the members (actors) of the Group
+func (g *Group) Members() []string {
+	return g.members
+}
+
+// Fastest ensures that the Broadcast returns the
+// BroadcastResult for the fastest member in the Group
+func (g *Group) Fastest() *Group {
+	return &Group{
+		members: g.members,
+		fastest: true,
+	}
+}
+
+// ExceptSuccesses filters out the successful members of the Group
+func (g *Group) ExceptSuccesses(res BroadcastResult) *Group {
+	var successes []string
+	for member, r := range res {
+		if r.Err == nil {
+			successes = append(successes, member)
+		}
+	}
+	return &Group{
+		fastest: false,
+		members: removeStrings(g.members, successes),
+	}
+}
```

Comment:
I think this function needs some mechanical sympathy. We already have the result set, which is a map where each key is the receiver.

Can we make `ExceptSuccesses` something like:

```go
func (g *Group) ExceptSuccesses(res BroadcastResult) *Group {
    ng := &Group{
        members: make([]string, 0, len(g.members), // Guarantee just one allocation.
    }

    for _, m := range g.members {
        // Check is member has a failure in the result set, in which case
        // add it to the new group so it can be operated on.
        if v := res[m]; v == nil || v.Err != nil {
            ng.members = append(ng.members, m)
        }
    }

    return ng
}
```

This way we iterate over the members just once, and we don't need any helper functions, so less code.